lib/oc-tests/new_locking/test_downloadFilePlusAction.py

```diff
@@ -0,0 +1,222 @@
+        },
+]
+
+@add_worker
+def setup(step):
+    step(1, 'setup')
+    reset_owncloud_account(num_test_users=config.get('accounts', 1))
+    reset_rundir()
+
+def downloader(step):
+    process_number = parse_worker_number(reflection.getProcessName())
+    user_account = sconf.oc_account_name if process_number <= 0 else '%s%i' % (sconf.oc_account_name, process_number)
+
+    step(2, 'create big file')
+
+    d = make_workdir()
+
+    list_files(d)
+    run_ocsync(d, user_num=None if process_number <= 0 else process_number)
+
+    # sync a big file
+    target_filename = os.path.join(d, 'folder', 'bigfile.dat')
+    mkdir(os.path.join(d, 'folder'))
+    mkdir(os.path.join(d, 'folder2'))
+    createfile(target_filename,'10',count=1000,bs=10000)
+    sum5 = md5sum(target_filename)
+
+    run_ocsync(d, user_num=None if process_number <= 0 else process_number)
+    list_files(d)
+
+    step(3, 'download file async')
+
+    tmpfile = tempfile.mkstemp()
+    # download the file asynchronously
+    download_thread = pyocaction(user_account, sconf.oc_account_password, True, 'get_file', '/folder/bigfile.dat', tmpfile[1], pyocactiondebug=True)
+
+    step(5, 'check result and cleanup')
+
+    # wait until the download finish
+    download_thread[0].join()
+
+    sum5_2 = md5sum(tmpfile[1])
+    # check both md5 matches
+    logger.debug('checking md5sum of the downloaded files')
+    error_check(sum5 == sum5_2, 'uploaded file is different than the downloaded file [%s] - [%s]' % (sum5, sum5_2))
+    # remove temporal file
+    os.remove(tmpfile[1])
+
+def doer(step):
+    method = config.get('action_method', 'put_file_contents')
+    args = config.get('action_args', ('/folder/bigfile.dat', '123'*50))
+    kwargs = config.get('action_kwargs', {})
+
+    process_number = parse_worker_number(reflection.getProcessName())
+    user_account = sconf.oc_account_name if process_number <= 0 else '%s%i' % (sconf.oc_account_name, process_number)
+
+    step(4, 'action over file')
+
+    # perform the action
+    try:
+        result = pyocaction(user_account, sconf.oc_account_password, False, method, *args, **kwargs)
+
+        step(4, 'check results')
+        # check successful result
+        logger.debug('check %s method finished correctly' % method)
+        error_check(result, method + ' action didn\'t finish correctly')
+
+        # perform extra check
+        check = config.get('extra_check', None)
+        if check:
+            logger.debug('additional check %s' % check)
+            check_params = config.get('extra_check_params', ())
+            error_check(globals()[check](*check_params), 'extra check failed: %s %s' % (check, check_params))
+    except owncloud.ResponseError as e:
+        logger.debug('%s action failed. Checking the status to know if the file is locked' % (method,))
+        error_check(e.status_code == 423, 'unexpected status code [%i] : %s' % (e.status_code, e.get_resource_body()))
```

Comment:
To clarify, you could have an initial sync loop that tries and sync a few times and catches 423. (Other exceptions would be interpreted as unexpected). Once no more exception is thrown, it means the download must have succeeded, so exit the loop and verify the downloaded file.